api/src/main/java/org/apache/iceberg/expressions/Literals.java

```diff
@@ -599,6 +600,11 @@ public String toString() {
     Object writeReplace() throws ObjectStreamException {
       return new SerializationProxies.FixedLiteralProxy(value());
     }
+
+    @Override
+    public String toString() {
+      return "X'" + BaseEncoding.base16().encode((toByteBuffer().array())) + "'";
```

Comment:
It is not safe to use `ByteBuffer.array` like this. Can you please use `ByteBuffers.toByteArray`?